tests/jballerina-unit-test/src/test/resources/test-src/statements/variabledef/global_variable_with_uninitialized_vars.bal

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2023 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+int x;
+int y = foo();
+
+function init() {
+    x = 2;
+}
+
+function foo() returns int => x;
```

Comment:
Can we add more examples like this to test different scopes? e.g.
- Use lamba function, 
- Use normal function body